neoload/commands/report.py

```diff
@@ -0,0 +1,1057 @@
+
+    fill_single_stats(__id, components, data)
+
+    fill_single_events(__id, components, data)
+
+    statistics_list = get_standard_statistics_list()
+
+    use_txn_raw = should_raw_transactions_data(__id, time_filter)
+
+    fill_single_requests(__id, elements_filter, time_binding, statistics_list, use_txn_raw, components, data)
+
+    fill_single_transactions(__id, elements_filter, time_binding, statistics_list, use_txn_raw, components, data)
+
+    fill_single_monitors(__id, components, data)
+
+    fill_single_ext_data(__id, components, data)
+
+    fill_single_controller_points(__id, components, data)
+
+    return data
+
+def get_standard_statistics_list():
+    return ["AVG_DURATION","MIN_DURATION","MAX_DURATION","COUNT","THROUGHPUT",
+                "ELEMENTS_PER_SECOND","ERRORS","ERRORS_PER_SECOND","ERROR_RATE",
+                "AVG_TTFB","MIN_TTFB","MAX_TTFB"]
+
+
+def fill_single_summary(__id, time_binding, time_filter, components, data):
+    if components['summary'] or components['slas'] or time_filter is not None:
+        gprint("Getting test results...")
+        json_result = rest_crud.get(get_end_point(__id))
+        json_result = add_test_result_summary_fields(json_result)
+        data['summary'] = json_result
+        if time_binding is not None:
+            time_binding["summary"] = json_result
+            time_binding = fill_time_binding(time_binding)
+    return time_binding
+
+def result_seems_corrupted(data):
+    return data['summary']['terminationReason'] in ['LG_AVAILABILITY', 'FAILED_TO_START','UNKNOWN']
+
+def fill_single_slas(__id, components, data):
+    if components['slas']:
+        status = data['summary']['status']
+        gprint("Getting global SLAs...")
+        data['sla_global'] = [] if status!='TERMINATED' else rest_crud.get(get_end_point(__id, __operation_sla_global))
+        gprint("Getting per-test SLAs...")
+        data['sla_test'] = [] if status!='TERMINATED' else rest_crud.get(get_end_point(__id, __operation_sla_test))
+        gprint("Getting per-interval SLAs...")
+        data['sla_interval'] = rest_crud.get(get_end_point(__id, __operation_sla_interval))
+
+def fill_single_stats(__id, components, data):
+    if components['statistics']:
+        gprint("Getting test statistics...")
+        data['statistics'] = rest_crud.get(get_end_point(__id, __operation_statistics))
+
+def fill_single_events(__id, components, data):
+    if components['events']:
+        gprint("Getting events...")
+        data['events'] = rest_crud.get(get_end_point(__id, __operation_events))
+
+def fill_single_requests(__id, elements_filter, time_binding, statistics_list, use_txn_raw, components, data):
+    if components['all_requests']:
+        gprint("Getting all-request data...")
+
+        json_elements_requests = rest_crud.get(get_end_point(__id, __operation_elements) + "?category=REQUEST")
+        json_elements_all_requests = list(filter(lambda m: m['id'] == 'all-requests', json_elements_requests))
```

Comment:
Is it the expected behavior ?
Because of the filter here, in the json output we always have only 1 request named 'all-requests' in the list...

Reply:
This is expected behavior (OK for Paul in today meeting). We don't need each request in the report.